source/obstacle/Obstacle.hx

```diff
@@ -1,57 +1,81 @@
 package obstacle;
 
+import flixel.FlxBasic;
 import flixel.FlxG;
 import flixel.FlxSprite;
 
-class Obstacle extends FlxSprite
-{
-	public function new(?parameters:ObstacleParameters)
-	{
+class Obstacle extends FlxSprite {
+	public function new(?parameters:ObstacleParameters) {
 		super();
 
 		immovable = true;
 
-		if (parameters != null)
-		{
+		if (parameters != null) {
+			// this gives the obstacle its init values
 			init(parameters);
 		}
 	}
 
-	public function init(parameters:ObstacleParameters)
-	{
+	public function init(parameters:ObstacleParameters) {
+		// revive the obstace to be drawn again
 		super.reset(parameters.x, parameters.y);
 
+		// set the velocity to whatever the speed is but neg
+		var upOrDown = FlxG.random.int(0, 1);
 		velocity.x = -parameters.speed;
 
+		if (upOrDown == 1) {
+			velocity.y = -(FlxG.random.float(parameters.speed / 2, parameters.speed));
+		} else {
+			velocity.y = (FlxG.random.float(parameters.speed / 2, parameters.speed));
+		}
+
+		// used to create as a rectangle
 		makeGraphic(parameters.width, parameters.height);
 	}
 
-	override public function update(elapsed:Float)
-	{
-		if (movedOffScreen())
-		{
+	override public function update(elapsed:Float) {
+		// if the object has moved off screen
+		if (movedOffScreen()) {
+			// dont draw it
 			kill();
 		}
 
+		// adds the bounce effect
+		/* if (y < 0 || y > FlxG.height - height) {
+			velocity.y = -velocity.y;
+		}*/
+
+		if (y < 0 - height) {
+			y = FlxG.height;
+		}
+
+		if (y > FlxG.height) {
+			y = 0 - height;
+		}
+
 		super.update(elapsed);
 	}
 
-	private function movedOffScreen()
-	{
+	private function movedOffScreen() {
+		// check to see if the object has moved off screen
 		return x + width < FlxG.camera.scroll.x;
 	}
+
+	public static function overlapsWithPlayer(player:FlxBasic, obstacle:DeadlyObstacle) {
+		player.kill();
+	}
```

Comment:
this shouldn't take DeadlyObstacle as its second parameter, and the functionality will need to change when you bring in master's system with new lives.

Also, I think the distinction between DeadlyObstacles and Obstacles becomes mostly meaningless with this change - if everything's deadly, we don't really need to have two different classes, right? The old difference was so that some objects could push and other objects could kill. At this point, they could be the same class with parameterized colors. 

I like having the option of having different kinds of obstacles that behave differently, so I'd be in favor of just making Obstacle` / `DeadlyObstacle` take color as a parameter and making all of the generators you set up be for `DeadlyObstacle`.  This method wouldn't be needed under that system, the overlap-to-kill will be fully in the realm of `DeadlyObstacle` functionality.

Reply:
Oh, I guess one significant difference is that this doesn't get rid of the `Obstacle` object, but that's going to create some problems with the functionality that's in main now - the idea is that the player takes damage from objects instead of being outright killed and reset.

If the obstacles flying across here aren't destroyed when they touch the player, then they'll deal damage every frame and pretty much always result in immediate death.